Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -226,13 +256,92 @@ public final class SecureEnclaveValet: NSObject {
 
     // MARK: Internal Properties
 
+    /// A provider of an `LAContext` used when needing to manually evaluate an authentication policy
+    /// prior to retrieving a value from the keychain in order to allow the evaluation prompt to show a
+    /// custom fallback button to the user.  Should always return a new instance of `LAContext` in
+    /// practice, but specified as an internal var to allow it to be overridden in tests.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    internal lazy var authenticationContextProvider: () -> LAContext = { LAContext() }
+    #endif
+
     internal let service: Service
 
     // MARK: Private Properties
 
     private let lock = NSLock()
     private let baseKeychainQuery: [String : AnyHashable]
 
+    // MARK: - Private Methods
+
+    /// Synchronously calls `LAContext.evaluatePolicy`, passing the evaluated context
+    /// via `kSecUseAuthenticationContext` when querying for the string at the given
+    /// key in the secure enclave.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    private func synchronouslyEvaluatePolicy(
+        forKey key: String,
+        withPrompt userPrompt: String,
+        withFallbackTitle fallbackTitle: String
+    ) throws -> String {
+        // Use a semaphore to block the thread and perform the evaluation synchronously.
+        let semaphore = DispatchSemaphore(value: 0)
+        var result: Result<String, Error>?
+        let policy = try accessControl.getPolicy()
+
+        let authenticationContext = authenticationContextProvider()
+        authenticationContext.localizedFallbackTitle = fallbackTitle
+        authenticationContext.evaluatePolicy(
```

Comment:
I've spent some time over the past few days trying to use `evaluateAccessControl` here instead of `evaluatePolicy` so we don't have to try to map from our access control to `LAPolicy`, but `evaluateAccessControl` doesn't use the `localizedFallbackTitle` 😞

Reply:
Welp. Good investigation at least!